packages/docusaurus-theme-common/src/components/Details/index.tsx

```diff
@@ -54,6 +54,16 @@ export function Details({
   // only after animation completes, otherwise close animations won't work
   const [open, setOpen] = useState(props.open);
 
+  let SummaryElement = null;
+
+  if (React.isValidElement(summary)) {
+    SummaryElement = summary;
+  } else if (typeof summary === 'string') {
+    SummaryElement = <summary>{summary}</summary>;
+  } else {
+    SummaryElement = <summary>Details</summary>;
+  }
```

Comment:
I suggest:

```suggestion
  const summaryElement = React.isValidElement(summary) ? summary : <summary>{summary ?? "Details"}</summary>;
```

(Don't apply this directly; you also need to format it and change the casing of the `SummaryElement` below. You also need the `eslint-disable` comment that was there)

Reply:
Good idea. Changed it accordingly.